src/app/pages/index.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { createStackNavigator } from '@react-navigation/stack';
+import { RouteNames } from '@common/types/navigation';
+import MainScreen from '@pages/MainArticlesScreen';
+import ArticleScreen from '@pages/ArticleScreen';
+import ProfileScreen from '@pages/ProfileScreen';
+import { ArticleData } from '@core/pages/MainArticlesScreen/namespace';
+import { NavigationTitle } from '@common/components';
+
+export type RootStackParamList = {
+  [RouteNames.MainScreen]: undefined;
+  [RouteNames.ArticleScreen]: { article: ArticleData };
+  [RouteNames.ProfileScreen]: undefined;
+};
+
+const Stack = createStackNavigator<RootStackParamList>();
+
+export const NavigationStructure = () => {
+  return (
+    <Stack.Navigator initialRouteName={RouteNames.MainScreen}>
+      <Stack.Screen
+        name={RouteNames.MainScreen}
+        component={MainScreen}
+        options={{
```

Comment:
Im wondering if these options should be placed there. If we will have complex app we will have liek 30-40 differnt routes here with different options diplsyaing different component in header and other parts.

What about moving options close to component and import and use here?

Reply:
I used a build-in feature that allows to pass shared screen options for all the routes. In case it can be extended with another options per route, imported directly from component. 

What do you think about it?

Reply:
Good idea, lets have it like you said.